lib/src/us/kbase/kbasefeaturevalues/KBaseFeatureValuesImpl.java

```diff
@@ -117,13 +119,20 @@ public void estimateK(EstimateKParams params,
                 params.getNeighbSize(), params.getMaxItems());
         provenance.get(0).withDescription("K estimation for K-Means clustering method")
                 .withInputWsObjects(Arrays.asList(params.getInputMatrix()));
-        getWsClient().saveObjects(new SaveObjectsParams().withWorkspace(params.getOutWorkspace())
+	WorkspaceClient wsc = getWsClient();
+        List<Tuple11<Long,String,String,String,Long,String,Long,String,String,Long,Map<String,String>>> getInfo = wsc.saveObjects(new SaveObjectsParams().withWorkspace(params.getOutWorkspace())
                 .withObjects(Arrays.asList(new ObjectSaveData()
                 .withType("KBaseFeatureValues.EstimateKResult").withName(params.getOutEstimateResult())
                 .withData(new UObject(toSave)).withProvenance(provenance))));
+
+	WorkspaceIdentity wi = new WorkspaceIdentity();
+	wi.setWorkspace(params.getOutWorkspace());
+	Tuple9 t9 = wsc.getWorkspaceInfo(wi);
+	String outref = t9.getE1() + "/" + getInfo.get(0).getE1() + "/" +  getInfo.get(0).getE5();
```

Comment:
getInfo already has the wsid in it, no need to call the ws again

Reply:
This applies to many places in this code - there's a ton of duplicated code here. DRYing it up would be good.

Reply:
The extra ws call has been factored out.